### PR TITLE
Retirement

### PR DESCRIPTION
**Pull request by gila**
The first commit provides a bit more visibility into why the creation of channels might have failed.

### CODE REVIEW

File: mayastor/src/bdev/nexus/nexus_channel.rs [hunk 1]
```diff
@@ -125,12 +125,12 @@ impl NexusChannelInner {
...
-                    BdevHandle::try_from(c.get_descriptor().unwrap()).unwrap(),
-                );
-                self.readers.push(
-                    BdevHandle::try_from(c.get_descriptor().unwrap()).unwrap(),
-                );
+                if let Ok(hdl) = c.handle() {
+                    self.readers.push(hdl);
+                } else {
+                    c.set_state(ChildState::Faulted(Reason::CantOpen));
+                    error!("failed to create handle for {}", c);
```
Comment (paulyoong) [inline mayastor/src/bdev/nexus/nexus_channel.rs:132]: Should this message be "failed to get handle..." as we aren't actually creating a handle?

File: mayastor/src/bdev/nexus/nexus_channel.rs [hunk 2]
```diff
@@ -139,13 +139,14 @@ impl NexusChannelInner {
...
-                            .unwrap(),
-                    )
-                })
-                .for_each(drop);
+                .for_each(|c| {
+                    if let Ok(hdl) = c.handle() {
+                        self.writers.push(hdl);
+                    } else {
+                        c.set_state(ChildState::Faulted(Reason::CantOpen));
+                        error!("failed to create handle for {}", c);
```
Comment (paulyoong) [inline mayastor/src/bdev/nexus/nexus_channel.rs:147]: Likewise, should this be "..get handle..."?

File: mayastor/src/core/descriptor.rs [hunk 3]
```diff
@@ -37,6 +37,10 @@ impl Descriptor {
     pub fn get_channel(&self) -> Option<IoChannel> {
         let ch = unsafe { spdk_bdev_get_io_channel(self.0) };
         if ch.is_null() {
+            error!(
+                "failed to get IO channel for, probably low on memory! {}",
```
Comment (paulyoong) [inline mayastor/src/core/descriptor.rs:41]: I think the curly braces were meant to be after the "for" i.e. "failed to get IO channel for{},"

**Review by paulyoong — APPROVED** [review]

**Review by jkryl — APPROVED** [review]

**Comment by gila** [comment]
Bors merge

**Pull request merged**